src/dbnode/client/connection_pool_test.go

```diff
@@ -47,9 +47,25 @@ var (
 
 type noopPooledChannel struct {
 	address string
+	closed  int32
+}
+
+func asNoopPooledChannel(c Channel) *noopPooledChannel {
+	cc, ok := c.(*noopPooledChannel)
+	if !ok {
+		panic("not a noopPooledChannel")
+	}
+	return cc
+}
+
+func (c *noopPooledChannel) Closed() bool {
+	return atomic.LoadInt32(&c.closed) == 1
+}
+
+func (c *noopPooledChannel) Close() {
+	atomic.StoreInt32(&c.closed, 1)
```

Comment:
Is it sensitive to double closing? Should we maybe do an increment here in case it is? Otherwise consider `atomic.Bool`.

Reply:
There is no atomic.Bool in std lib.

Reply:
But the question whether we should check against double closing still applies.